src/dotnet-interactive-vscode/src/ipynbUtilities.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { getNotebookSpecificLanguage } from "./interactiveNotebook";
+
+// the shape of this is meant to match the cell metadata from VS Code
+interface CellMetadata {
+    custom?: { [key: string]: any } | undefined,
+}
+
+export interface DotNetCellMetadata {
+    language: string | undefined,
+}
+
+function isDotNetCellMetadata(arg: any): arg is DotNetCellMetadata {
+    return arg
+        && typeof arg.language === 'string';
+}
+
+export function getDotNetMetadata(metadata: CellMetadata | undefined): DotNetCellMetadata {
+    if (metadata &&
+        metadata.custom &&
+        metadata.custom.metadata &&
+        metadata.custom.metadata.dotnet_interactive &&
+        isDotNetCellMetadata(metadata.custom.metadata.dotnet_interactive)) {
```

Comment:
can't we use `isDotNetCellMetadata(metadata?.custom?.metadata?.dotnet_interactive)`

Reply:
That works for the check, but it makes the assignment inside the `if`-block more difficult.  By doing it this way the TypeScript compiler lets me simply say `return metadata.custom.metadata.dotnet_interactive;` instead of `return <DotNetCellMetadata>metadata?.custom?.metadata?.dotnet_interactive;`.